common/source/docs/common-brushless-escs.rst

```diff
@@ -36,18 +47,97 @@ The DShot ESC protocol's key advantages are:
+
+The current commands supported are:
+
+-    Dshot LEDs on/off
+-    Dshot Buzzer on/off
+-    Reverse motor direction
+-    Enable 3D operation
+
+.. warning:: Currently, ArduPilot supports the command set (:ref:`SERVO_DSHOT_ESC<SERVO_DSHOT_ESC>` =1) that is commonly used, however, others are appearing and may not be compatible, resulting in undefined operation. Use caution (remove blades!) until correct operation using type=1 is verified for your ESC
 
-A `search for "BLHeli32 shopping" <https://www.google.com/search?q=blheli32&tbm=shop>`__ turns up many compatible ESCs.  Look for an ESC which includes the telemetry wire connector like the `HolyBro Tekko32 shown below <https://shop.holybro.com/holybro-tekko32-esc35a_p1074.html>`__
+Reversible DShot ESCs
+---------------------
+
+Currently, only BLHeli32 and BLHeli-S capable reversible DShot ESCs are supported. In order to use one, the output which drives it must be designated with the appropriate bit in the :ref:`SERVO_BLH_3DMASK<SERVO_BLH_3DMASK>` bitmask parameter. This will map the outputs 1000-1500-2000 values to the correct digital values for the ESC to provide FullReverse-Idle-FullForward range operation, respectively.
+
+If the craft has been setup for DShot commands then ArduPilot will supply the correct command at startup in order to set the ESCs in reversible mode.
+
+In a similar fashion, normal output rotation direction can be reversed by setting :ref:`SERVO_BLH_RVMASK<SERVO_BLH_RVMASK>` without any changes needing to be made through ESC setup software (e.g. BLHeliSuite). This can also be used on ESCs with forward and reversed active operation, ie reversible ESCs, to set the "forward" direction's rotation.
+
+.. note:: Currently, ArduPilot only supports the use of reversible ESCs for Plane and Rover, not Copter.
+
+.. _output-groups:
+
+Autopilot Output Capabilities
+=============================
+
+While all the servo/motor outputs of an ArduPilot autopilot are capable of Normal PWM operation at 50Hz frame rates, not all are capable of other ESC protocol configurations. And, usually, these configurations must apply to pre-designated groups of outputs, even if they are not all driving an ESC. So the following cautions apply:
+
+#. The 8 "MAIN" outputs of autopilots using an IOMCU (like PixHawk and Cube), cannot be used for  protocols other than Normal PWM and OneShot. On these autopilots, only the additional "AUX" outputs can properly support OneShot125 and DShot protocols. If you attempt to set a "MAIN" output to DShot, then normal PWM output will occur, even though it has been set to a DShot protocol.
```

Comment:
I think we should change this section's title from "Auto Output Capabilities" to something more specific like, "Restrictions on which PWM outputs can be used".  We might want to have a link in each section (there should be one section per protocol above) warning users that they need to come down and read this general section.

Reply:
I think thats a bit too long....how about Mixing ESC protocols?

Reply:
That's better.. "Mixing PWM protocols"?.. either is fine though.

By the way, I think this line , "only the additional "AUX" outputs can properly support OneShot125 and DShot protocols" could be expanded to mention BLHeli, "only the additional "AUX" outputs can properly support OneShot125, DShot and BLHeli protocols" unless you think users will really know that by the time they get to this point in the docs that BLheli is a superset of Dshot.